### ISSUE TITLE
Webhook signin request doesn't works using express code example

### ISSUE THREAD

**Issue by Romakita** [stripe/stripe-node#1886]
### Describe the bug

Hello team!

I'm facing to an issue using the constructEvent method in nodejs.

I used your given code example but I always the same error message:

```
Error message: No signatures found matching the expected signature for payload. Are you passing the raw request body you received from Stripe? 
```

I tested using the Koa and Express code example given here:
https://github.com/stripe/stripe-node/blob/master/examples/webhook-signing/express/package.json
https://github.com/stripe/stripe-node/blob/master/examples/webhook-signing/koa/package.json

### To Reproduce

I created a repository to reproduce issue over express/koa/Ts.ED:

https://github.com/Romakita/tsed-stripe-issue

In the readme you have the steps reproduction for each framework:

## Start

Add `.env.local` file with the followings lines:

```dotenv
STRIPE_SECRET_KEY=pk_test_5***
STRIPE_WEBHOOK_SECRET=sk_test_5***
```

### Ts.ED

```
yarn start
yarn stripe:forward:tsed
yarn stripe:trigger
```

### Express

```
yarn start:express
yarn stripe:forward:express
yarn stripe:trigger
```

### Koa

```
yarn start:koa
yarn stripe:forward:koa
yarn stripe:trigger
```

### Issue

All example return the followings:

```
Error message: No signatures found matching the expected signature for payload. Are you passing the raw request body you received from Stripe? 
```

### Expected behavior

The webhook request works as expected

### Code snippets

```JavaScript
https://github.com/Romakita/tsed-stripe-issue
```

### OS

macOs

### Node version

Node v18.15.0

### Library version

stripe-node v13.3.0

### API version

2023-08-16

**Comment by remi-stripe** [comment]
@Romakita Webhook signature verification with Node.js can be quite finicky as it's fairly subtle based on a lot of variables around your environment set up. I recommend reading through https://github.com/stripe/stripe-node/issues/341 which has dozens of potential solutions from other developers. 

If that doesn't help, then I recommend reaching out directly to our support team for help as this is more an integration support question: https://support.stripe.com/contact

**Comment by Romakita** [comment]
Hello  @remi-stripe

Thanks for your answers. But I think you haven't taken a time to read my issue ;). I reproduce the bug with your code example! simply by checkout your code here:
https://github.com/stripe/stripe-node/blob/master/examples/webhook-signing/express

It doesn't works, this is why I opened an issue, because many other devs take your code example and expect that will works just by copy past his stripe apiKey and Secret. But isn't the case here.

I wasted my time to give you repository example that reproduce the bug step by step with your Express/Koa code example (there is no change) and with my Ts.ED framework.

I don't expect from you a response around the Ts.ED framework (but my consumer framework expect an answer ^^ from me), but I expect that your example works out of the box. I would like your repository to give a working example!

I already integrated stripe service 3 or 4 years ago without issue. Your public nodejs API doesn't seems to have changed since that. 

During this time I will read and reread the issue that you transmitted to me and on which I must have come across before.

See you
Romain

**Comment by remi-stripe** [comment]
@Romakita I did read the report you provided but this still looks a lot more integration support to us. I don't think you wasted your time giving us a reproduction since it will help our support team work with you to reproduce the issue and find the root cause. If a fix is needed in this repo or examples we can merge it afterwards!

**Comment by Romakita** [comment]
Yes it seems to be an integration issue. But actually, the code example from your repo doesn't works. The calculted hash differ from the given header signature. But the example give a Buffer as it's expected by the constructEvent method. 

So my hypotheses are either the buffer provided by koa or express are not correct, or it is linking to the crypto part under nodejs 18 which would be incorrect or both at the same time.

See you ;) and thanks for answer!

**Comment by richardm-stripe** [comment]
By the way, https://github.com/Romakita/tsed-stripe-issue does not appear publicly accessible?

**Comment by richardm-stripe** [comment]
I was able to successfully verify webhook signatures by running our examples using `stripe listen`.

@Romakita If you're on a recent stripe-node, judging from the fact that your error message doesn't seem to contain [the text](https://github.com/stripe/stripe-node/blob/fe81d1f28064c9b468c7b380ab09f7a93054ba63/src/Webhooks.ts#L370C14-L370C73) about receiving a "parsed javascript object", I think it's probably not an issue with Koa/Express/your Javascript framework and is more likely related to the secret. It might be worth double-checking that you are passing the correct webhook signing secret into `constructEvent`, and that the events coming into your application are the ones you expect (e.g. make sure you're not using a livemode secret for testmode events, that sort of thing)

**Comment by Romakita** [comment]
Hello @richardm-stripe 
I changed the visibility repo. I forgot that github make new repository in private visibility first.

Using the `stripe listen` it works. I see the secret key isn't the same as the secret key given by the backoffice here:

<img width="1191" alt="Capture d’écran 2023-08-30 à 07 41 21" src="https://github.com/stripe/stripe-node/assets/1763311/d5cc4b72-efa7-46e1-8e1e-ac7ffda680b7">

The webhook secret start by `ws_`. I admit that I missed that. I mechanically took the secret key, And I haven't seen that the `stripe forward` give this information. My apologize guys. 

I'll edit the official document the Ts.ED website over stripe integration, to avoid this mistake when we use the `stripe forward` command.

Thanks for your time and sorry about this issue.
Have a nice day @remi-stripe  @richardm-stripe 

See you

**Comment by mbeato** [comment]
late reply for searchers landing here — if you hit "No signatures found matching the expected signature" with the express example, the most common cause is the wrong secret type:

- `whsec_*` → webhook signing secret (dashboard → developers → webhooks → reveal)
- `sk_*` → api secret
- `pk_*` → publishable key

they're not interchangeable. looking at the `.env` snippet up-thread, the `STRIPE_WEBHOOK_SECRET=sk_test_*` would produce exactly this error — webhook secret should start with `whsec_`.

second-most-common: body parsed by `express.json()` before the verify route runs. fix is `app.use('/webhook', express.raw({type:'application/json'}), handler)` mounted *before* `express.json()`.

made [stripesig](https://stripesig.apimesh.xyz?utm_source=github&utm_medium=issue-comment&utm_campaign=stripe-node-1886) that runs both checks automatically — paste payload + secret + signature header, get a plain-english diagnosis. free, no signup.
